File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/java/F04.P3.java
import java.util.List;

class Main {
    static final class Box {
        int id;

        Box(int id) {
            this.id = id;
        }
    }

    static boolean probe() {
        // BEGIN PROBE F04.P3
        Box first = new Box(5);
        Box second = List.of(first).get(0);
        boolean same = first == second;
        return same;
        // END PROBE F04.P3
    }

    public static void main(String[] args) {
        System.out.println(probe());
    }
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/kotlin/F04.P3.kt
class Box(var id: Int)

fun probe(): Boolean {
    // BEGIN PROBE F04.P3
    val first = Box(5)
    val second = listOf(first)[0]
    val same = first === second
    return same
    // END PROBE F04.P3
}

fun main() {
    println(probe())
}
